### DISCUSSION TITLE
admin list gives an error

### DISCUSSION THREAD

**Discussion by jsaathof** [smallstep/certificates#1579] [Q&A]
Hi,

I have been running for a while now with step-ca for ACME. I wanted to add _eab_ but I got an error. I was asked for an admin and password and encountered an error after entering them.
```
✔ Please enter admin name/subject (e.g., name@example.com): admin@test.org
✔ Provisioner: test (ACME)
error creating admin client: step ACME provisioners do not support token auth flows
```
I had removed all JWK configuration and did not know that I do not have an admin now.
For testing I have installed step and step-cli on a new server for testing. This was an empty configuration with an admin. But when I try to add or even list the admins I got a new error.
```
[build]:~
root# step ca admin list
No admin credentials found. You must login to execute admin commands.
✔ Please enter admin name/subject (e.g., name@example.com): admin@test.org
✔ Provisioner: admin@test.org (JWK) [kid: wDDZYs_NDOetIpGNjx9GLp_wFsnt3pfy7_EkrXpHdDw]
Please enter the password to decrypt the provisioner key:
json: cannot unmarshal number into Go value of type ca.AdminClientError
```
I have searched the internet for this error but did not find a real answer. I found this: https://github.com/smallstep/certificates/discussions/963 but it was not clear to me on what to do to fix this. I am running both step and step-ca on the same machine. And I do not want to change to a different database.

Bear in mind this is an empty setup and I do not understand what I have configured wrong. This is supposed to work, isn't it?

**Comment by udf2457** [comment]
@jsaathof did you ever find a solution to this ? 

I have tried e.g. `--admin-provisioner="Admin JWK" --issuer=myacmename` but that doesn't work either ?

**Reply by rhubear25** [reply]
I was gonna say, the PC command Ive used to get the Step CA server list of admins is --

 step ca provisioner list
 
I do have remote management enabled on the Docker compose --
     environment:
      # CA Identity
      - DOCKER_STEPCA_INIT_NAME=SmallstepCA
      - DOCKER_STEPCA_INIT_DNS_NAMES=localhost,xx.xx.xx.xx
      - DOCKER_STEPCA_INIT_REMOTE_MANAGEMENT=true
      # Admin Setup
      - DOCKER_STEPCA_INIT_PROVISIONER_NAME=admin
      - DOCKER_STEPCA_INIT_PASSWORD_FILE=/run/secrets/ca-password
      # Features
      - DOCKER_STEPCA_INIT_ACME=true
      - DOCKER_STEPCA_INIT_SSH=true

My problem is updating the admin provisioner.  I'm on another thread with that.

**Comment by hslatman** [comment]
@jsaathof I know this is an old issue, but did you enable remote administration when creating the new CA? We're aware the error is not clear. The `step` CLI is trying to reach the admin API endpoint, which doesn't exist if remote administration is not enabled. Some docs on this subject are available here: https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management. 

With remote provisioner management, the provisioners are stored in the same database that your CA already uses for other records it keeps. `step ca admin` is only supported with remote provisioner management, as it isn't something you need if remote provisioner management is not enabled, as it's primarily functionality is managing the admins that can perform remote administration.

**Comment by hslatman** [comment]
@udf2457 `--admin-provisioner` and `--provisioner` should be usable. Those go combined with `step` as the default admin user subject, which uses the first JWK provisioner that was created as the "admin provisioner".

**Comment by udf2457** [comment]
Thanks @hslatman .  I'm busy trying to get device-attest-01 working at the moment (#2486) but once I've cracked that I'll return to this.   I suspect I probably forgot a CLI flag.
